### PR TITLE
[MTDSA-20961] [CK] Upgrade to play 3.0.

### PR DESCRIPTION
**Pull request by sudoku007**
Upgrade to play 3.0.

### CODE REVIEW

File: app/shared/controllers/RewriteableAssets.scala [hunk 1]
```diff
@@ -67,7 +68,7 @@ class RewriteableAssets @Inject() (errorHandler: HttpErrorHandler, meta: AssetsM
       case Some((assetInfo, acceptEncoding)) =>
         val connection = assetInfo.url(acceptEncoding).openConnection()
         // Make sure it's not a directory
-        if (Resources.isUrlConnectionADirectory(connection)) {
+        if (Resources.isUrlConnectionADirectory(this.getClass.getClassLoader, connection)) {
```
Comment (mattstephens-hmrc) [inline app/shared/controllers/RewriteableAssets.scala:71]: probably doesn't need the "this."

File: app/shared/routing/VersionRoutingRequestHandler.scala [hunk 2]
```diff
@@ -35,7 +35,7 @@ class VersionRoutingRequestHandler @Inject() (versionRoutingMap: VersionRoutingM
     extends DefaultHttpRequestHandler(
       webCommands = new DefaultWebCommands,
       optDevContext = None,
-      router = versionRoutingMap.defaultRouter,
+      router = ()=> versionRoutingMap.defaultRouter,
```
Comment (mattstephens-hmrc) [inline app/shared/routing/VersionRoutingRequestHandler.scala:38]: needs a reformat?

File: project/build.properties [hunk 3]
```diff
@@ -1 +1 @@
-sbt.version=1.7.2
+sbt.version=1.9.7
```
Comment (mattstephens-hmrc) [inline project/build.properties:1]: Just realised, the readme still specifies 1.6.x
Maybe change in the readme to "the sbt version specified in project/build.properties" ?

File: test/shared/connectors/ConnectorSpec.scala [hunk 4]
```diff
@@ -152,35 +152,35 @@ trait ConnectorSpec extends UnitSpec with Status with MimeTypes with HeaderNames
 
     protected lazy val requiredHeaders: Seq[(String, String)] = requiredDesHeaders
 
-    MockedAppConfig.desBaseUrl returns baseUrl
+    MockedAppConfig.desBaseUrl returns this.baseUrl
```
Comment (mattstephens-hmrc) [inline test/shared/connectors/ConnectorSpec.scala:155]: I don't think any of these need the "this.", worth removing?

File: app/shared/controllers/RewriteableAssets.scala (hunk 1 above)
Comment (sudoku007) [reply app/shared/controllers/RewriteableAssets.scala]: removed "this".

File: app/shared/routing/VersionRoutingRequestHandler.scala (hunk 2 above)
Comment (sudoku007) [reply app/shared/routing/VersionRoutingRequestHandler.scala]: done reformatting.

File: project/build.properties (hunk 3 above)
Comment (sudoku007) [reply project/build.properties:1]: updated readme with sbt 1.9.7

File: test/shared/connectors/ConnectorSpec.scala (hunk 4 above)
Comment (sudoku007) [reply test/shared/connectors/ConnectorSpec.scala:155]: "this" reference is needed here as baseUrl is shadowed and scala compiler throws error without "this."

**Review by mattstephens-hmrc — APPROVED** [review]

**Pull request merged**